torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py

```diff
@@ -76,11 +76,38 @@ def forward(self, *args, **kwargs):
             with save_on_cpu(pin_memory=True):
                 return self._checkpoint_wrapped_module(*args, **kwargs)
         else:
-            return self.checkpoint_fn(  # type: ignore[misc]
-                self._checkpoint_wrapped_module,
-                *args,
-                **kwargs
-            )
+            # Support keyword arguments for reentrant checkpoint. Note that this
+            # only works if user has specified self.checkpoint_impl and is not
+            # using their own custom checkpoint_fn.
+            if self.checkpoint_impl == CheckpointImpl.REENTRANT and kwargs != {}:
```

Comment:
debating whether we need the `kwargs != {}` or if we should just do this anyways even if there are no kwargs.

Reply:
I think the `kwargs != {}` is fine. Otherwise, we construct a new function at runtime unnecessarily (among some other unnecessary ops). Is there any reason to use this code path even when `kwargs` is empty?